snakemake/executors/__init__.py

```diff
@@ -516,11 +518,21 @@ def cached_or_run(self, job, run_func, *args):
         if to_cache:
             self.workflow.output_file_cache.store(job)
 
-    def shutdown(self):
+    def shutdown(self, jobids=None):
+        if jobids is None:
+            return
+        logger.debug("CPUExecutor shutting down, will kill {} jobs".format(len(jobids)))
+        # kill all jobs!
+        for jobid in jobids:
+            logger.debug("Killing job {}".format(jobid))
+            shell.kill(jobid=jobid, timeout=self.timeout)
+            logger.debug("Killed job {}".format(jobid))
+        logger.debug("Shutting down the thread pool")
```

Comment:
So, actually not killing jobs upon local failure (but instead finish all currently running without scheduling new ones) was intentional, and is also in line with what happens in cluster mode. However, I can understand that it is sometimes interesting to have a different behavior (which one can already get by hitting ctrl+c after the first failure is reported). So, how about adding this behavior as an optional flag `--kill-upon-failure`?